static/js/controllers/edit-user.js

```diff
@@ -82,34 +88,59 @@
         Select2Search($('#edit-user-profile [name=facility]'), ContactSchema.getPlaceTypes());
       });
 
-      var validatePassword = function() {
+      var validateRequired = function(fieldName, fieldDisplayName) {
+        if (!$scope.editUserModel[fieldName]) {
+          $scope.errors[fieldName] = $translate.instant('field is required', {
+            field: $translate.instant(fieldDisplayName)
+          });
+          return false;
+        }
+        return true;
+      };
+
+      var validatePasswordForEditUser = function() {
         var newUser = !$scope.editUserModel.id;
         if (newUser) {
-          if (!$scope.editUserModel.password) {
-            $scope.errors.password = $translate.instant('field is required', {
-              field: $translate.instant('Password')
-            });
-            return false;
-          }
+          return validatePasswordAndConfirm();
+        }
+
+        // if existing user : needs both fields, or none
+        if ($scope.editUserModel.password) {
+          return validatePasswordAndConfirm();
         }
-        if ($scope.editUserModel.password &&
-            $scope.editUserModel.password !== $scope.editUserModel.passwordConfirm) {
+        if ($scope.editUserModel.passwordConfirm) {
           $scope.errors.password = $translate.instant('Passwords must match');
```

Comment:
...or perhaps not, but why not just call `validatePasswordAndConfirm()` in any case?  It looks like you could replace 108-115 with:
```
if ($scope.editUserModel.password || $scope.editUserModel.password) {
  return validatePasswordAndConfirm();
}
return true;